app/src/main/java/com/termux/api/apis/BluetoothScanAPI.java

```diff
@@ -0,0 +1,122 @@
+package com.termux.api.apis;
+
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothDevice;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.util.JsonWriter;
+
+import com.termux.api.util.ResultReturner;
+import com.termux.shared.logger.Logger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class BluetoothScanAPI {
+
+    private static final ArrayList<BluetoothDevice> devices = new ArrayList<>();
+    private static final String LOG_TAG = "BluetoothScanAPI";
+    private static BluetoothAdapter adapter;
+    private static boolean scanning = false;
+
+    private static final BroadcastReceiver receiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (Objects.equals(intent.getAction(), BluetoothDevice.ACTION_FOUND)) {
+                BluetoothDevice device = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
+                if (device != null) {
```

Comment:
Any reason we'd expect an `ACTION_FOUND` with `device == null`? Add a comment if so, log an error saying "that shouldn't happen it but" did if not.